### PR TITLE
Some cleanup

### PR DESCRIPTION
**Pull request by kand**
Updates to components touched by `tunic-cms` `campaign-edit-form` in https://github.com/theonion/tunic-cms/pull/9.

### New
1. `cms-components` `$render.renderToRoot` now takes an `template` parameter that is the template string to use to render the given route.
1. Added `cms-notifications` directive that can be used to display notifications registered with `NotificationsService`.
1. Added `cms-tooltip` directive that can be applied as an attribute to an element that should open up a tooltip.
1. Added `cms-unsaved-changes-guard` directive that can be applied as an attribute on a form to intercept navigation when the form's `$dirty` property is `true` property.
1. `error`, `warning`, and `info` color palettes added to `colors.scss`.
1. `input-with-icon-container` added to `mixins.scss` for styling an input that looks like it has icons inside of it.
1. `button-unstyled` added to `mixins.scss` for removing all styling from a button element.

#### Image Upload
1. If nested within an angular `form` will set the `form` to dirty when any image changes occur.
1. Added `betty-editable-button-text` attribute which allows customization of text displayed on main button.
1. Added `betty-editable-input-name` attribute which opens up the angular form interface for this form component.

#### Button
1. Added `button-type` attribute which maps directly to `type` attribute on rendered button.
1. Added `button-disabled` attribute which will disable the button when truthy.
1. Added `button-glyph-class` attribute which specifies the class to use to style the rendered button.
1. Added `button-glyph-prefix` attribute which specifies the prefix to use for glyph.

#### Input
1. Added `cms-input-errors-show-when` attribute to turn all errors on/off for input.
1. Added `cms-input-errors` attribute that can be used to provide a map of error messages and when they should render.
1. Use `invalid` class on input to style it as invalid.

### Updated
1. `cms-nav-user` directive is now responsible for showing/hiding itself based on whether or not given user is logged in.
1. `cms-row` `cms-row-col-*` classes added for determining how many columns a given `cms-row` should take up in their flex container.

#### Image Upload
1. Buttons are now `type="button"` to prevent issues with form submission.
1. Crop changes update image display within the component.
1. `image-crop-modal` buttons are now `type="button"` to prevent issues with form submission.

#### Button
1. `glyphpos` can be set to `after` to render icon after component's transclusion.
1. `type` attribute defaults to `friendly`.

### Breaking
1. Removed `cms-layout` `cms-row` transclusion styling that was causing layout headaches.
1. Removed unused `TokenAuthConfig` `setAfterLoginPath` and `getAfterLoginPath`.

### CODE REVIEW

**Comment by kand** [comment]
@collin @spra85 there's a ton of stuff here, but please start reviewing. Will add to CHANGELOG once I verify `tunic-cms` campaign form is still working.

File: components/betty-cropper/betty-cropper.scss [hunk 1]
```diff
@@ -1,52 +1,47 @@
...
+      bottom: auto;
+      top: 10px;
+    }
+    .remove {
+      bottom: auto;
+      top: 10px;
+    }
+    .caption-alt {
+      position: absolute;
+      bottom: 0;
```
Comment (spra85) [inline components/betty-cropper/betty-cropper.scss]: Waiting for one of these @kand.  Can you alphabetize?

**Comment by spra85** [comment]
Looks fine so far.  We should try best to bite off smaller chunks on than this kind of wholesale refactoring if it's possible to break up.  Maybe do betty-editable, PR, ship.  Move on. etc.  If that's unavoidable, I think having the person working on this walk through the PR step by step over 15-20 min would be a big help prior to reviewing.

**Comment by collin** [comment]
:+1:s all around

**Pull request merged**